examples/ref-templates/assets/htlc.py

```diff
@@ -0,0 +1,100 @@
+
+    # Next, check that this is a payment transaction. 
+    pay_check = Txn.type_enum() == TxnType.Payment
+
+    # Next, check that the Receiver field for this transaction is empty 
+    # Because this contract can approve transactions that close out its entire balance, 
+    # it should never have a receiver.
+    rec_field_check = Txn.receiver() == Global.zero_address()
+
+    # Next, check that the Amount of algos transferred is 0. This is for the same reason as 
+    # above: we only allow transactions that close out this account completely, which 
+    # having a non-zero-address CloseRemainderTo will handle for us.
+    amount_check = Txn.amount() == Int(0)
+
+    # fold all the above checks into a single boolean.
+    required_condition = And(
+        fee_check,
+        pay_check,
+        rec_field_check,
+        amount_check
+    )
+
+    """Payout scenarios : At this point in the execution, there is one boolean variable on the 
+    stack that must be true in order for the transaction to be valid. The checks we have done 
+    above apply to any transaction that may be approved by this script.We will now check if we
+     are in one of the two payment scenarios described in the functionality section."""
+
+    # Scenario 1: Hash preimage has been revealed
+    # First, check that the CloseRemainderTo field is set to be the TMPL_RCV address.
+    recv_field_check = Txn.close_remainder_to() == TMPL_RCV
+
+    # Next, we will check that arg_0 is the correct preimage for TMPL_HASHIMG under TMPL_HASHFN. 
+    preimage_check = TMPL_HASHFN(Arg(0)) == Bytes("base64", TMPL_HASHIMG)
+
+    #Fold the "Scenario 1" checks into a single boolean.
+    scenario_1 = And(recv_field_check, preimage_check)
+
+
+    # Scenario 2: Contract has timed out
+    # First, check that the CloseRemainderTo field is set to be the TMPL_OWN address 
+    # (presumably initialized to be the original owner of the funds).
+    owner_field_check = Txn.close_remainder_to() == TMPL_OWN
+
+    # Next, check that this transaction has only occurred after the TMPL_TIMEOUT round.
+    timeout_check = Txn.first_valid() > Int(TMPL_TIMEOUT)
+
+    #Fold the "Scenario 2" checks into a single boolean.
+    scenario_2 = And(owner_field_check, timeout_check)
+
+    # At this point in the program's execution, the stack has three values. At the base of the 
+    # stack is a boolean holding the results of the initial transaction validity checks. 
+    # This is followed by two booleans indicating the results of the scenario 1 and 2 checks.
+
+    # We want to approve this transaction if we are in scenario 1 or 2. 
+    # So we logically OR the results of those checks together.
+    # Finally, we logically AND the scenario checks with the initial checks.
+    return And(Or(scenario_1, scenario_2), required_condition)
+
+    # At this point, the stack contains just one value: a boolean indicating 
+    # whether or not it has been approved by this contract.
```

Comment:
please move the last 2 lines just above return (combine it with the previous comments)

Reply:
ok